### PR TITLE
Topography fix for mpp io

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
Allows for backwards compatibility with mpp io

Fixes #505

**How Has This Been Tested?**
Tested on gaea with intel compiler, but fails in travis with use_mpp_io flag during linking

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: topography/topography.F90 [hunk 1]
```diff
@@ -97,11 +101,12 @@ module topography_mod
                          water_file = 'DATA/navy_pctwater.data'
    namelist /topography_nml/ topog_file, water_file
 ! </NAMELIST>
-
+#ifndef use_mpp_io
    integer, parameter    :: TOPOG_INDEX = 1
    integer, parameter    :: WATER_INDEX = 2
```
Comment (thomas-robinson) [inline topography/topography.F90:106]: Put the `#ifndef` below this.  These are namelist variables, so they need to be used regardless of IO

File: topography/topography.F90 [hunk 2]
```diff
@@ -787,91 +791,727 @@ subroutine interp_topog_2d ( blon, blat, zout, flag )
...
- integer :: j
+! <FUNCTION NAME="get_topog_stdev">
 
-   js = 1
-   do j = 1, size(ydat(:))-1
-      if (ybeg >= ydat(j) .and. ybeg <= ydat(j+1)) then
-         js = j
-         exit
-      endif
-   enddo
```
Comment (thomas-robinson) [inline topography/topography.F90:804]: Is this routine actually deleted?  You shouldn't delete any routines

Comment (rem1776) [reply topography/topography.F90:804]: It's still there, just outside the ifdef's

https://github.com/rem1776/FMS/blob/5d11c99917dce488a81cd90fc1b40824335dd48d/topography/topography.F90#L1369

**Comment by colingladueNOAA** [comment]
Is this ready for review?

**Comment by rem1776** [comment]
@colingladueNOAA Yes, passed with travis and just tested on orion with awg and intel

**Pull request merged**